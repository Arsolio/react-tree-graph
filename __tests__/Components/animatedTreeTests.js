import React from 'react';
import { shallow } from 'enzyme';

import AnimatedTree from '../../src/components/animatedTree';

describe('<AnimatedTree>', () => {
	test('renders correctly', () => {
		const props = {
			data: {
				name: 'Colour',
				childs: [{
					name: 'Black'
				}]
			},
			// Must not use children prop due to special meaning in react causing shallow renderer errors
			getChildren: d => d.childs,
			height: 100,
			width: 200
		};

		const tree = shallow(<AnimatedTree {...props}/>);

		expect(tree).toMatchSnapshot();
	});

	test('default getChildren looks at children prop', () => {
		const node = {
			children: [{
				name: 'child'
			}]
		};

		const children = AnimatedTree.defaultProps.getChildren(node);

		expect(children).toBe(node.children);
	});
});