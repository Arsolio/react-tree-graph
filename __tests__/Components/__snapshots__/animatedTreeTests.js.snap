// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedTree> renders correctly 1`] = `
<Animated
  duration={500}
  easing={[Function]}
  gProps={
    Object {
      "className": "node",
    }
  }
  getChildren={[Function]}
  height={100}
  keyProp="name"
  labelProp="name"
  links={
    Array [
      Object {
        "source": Node {
          "children": Array [
            Node {
              "data": Object {
                "name": "Black",
              },
              "depth": 1,
              "height": 0,
              "parent": [Circular],
              "x": 40,
              "y": 40,
            },
          ],
          "data": Object {
            "childs": Array [
              Object {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 40,
          "y": 10,
        },
        "target": Node {
          "data": Object {
            "name": "Black",
          },
          "depth": 1,
          "height": 0,
          "parent": Node {
            "children": Array [
              [Circular],
            ],
            "data": Object {
              "childs": Array [
                Object {
                  "name": "Black",
                },
              ],
              "name": "Colour",
            },
            "depth": 0,
            "height": 1,
            "parent": null,
            "x": 40,
            "y": 10,
          },
          "x": 40,
          "y": 40,
        },
      },
    ]
  }
  nodeProps={Object {}}
  nodeShape="circle"
  nodes={
    Array [
      Node {
        "children": Array [
          Node {
            "data": Object {
              "name": "Black",
            },
            "depth": 1,
            "height": 0,
            "parent": [Circular],
            "x": 40,
            "y": 40,
          },
        ],
        "data": Object {
          "childs": Array [
            Object {
              "name": "Black",
            },
          ],
          "name": "Colour",
        },
        "depth": 0,
        "height": 1,
        "parent": null,
        "x": 40,
        "y": 10,
      },
      Node {
        "data": Object {
          "name": "Black",
        },
        "depth": 1,
        "height": 0,
        "parent": Node {
          "children": Array [
            [Circular],
          ],
          "data": Object {
            "childs": Array [
              Object {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 40,
          "y": 10,
        },
        "x": 40,
        "y": 40,
      },
    ]
  }
  pathProps={
    Object {
      "className": "link",
    }
  }
  steps={20}
  svgProps={Object {}}
  textProps={Object {}}
  width={200}
/>
`;
